server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -317,18 +325,49 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    public void acquireLock(long primaryTerm, long generation, String resourceID) throws IOException {
+        Optional<String> metadataFile = getMetadataFileForCommit(primaryTerm, generation);
+        if (metadataFile.isEmpty()) {
+            String errorString = "Metadata file is not present for given primary term "
+                + primaryTerm + " and generation " + generation;
+            throw new FileNotFoundException(errorString);
+        }
+        mdLockManager.acquire(mdLockManager.getLockInfoBuilder().withMetadataFile(metadataFile.get())
+            .withResourceId(resourceID).build());
+    }
+
+    // Visible for testing
+    Optional<String> getMetadataFileForCommit(long primaryTerm, long generation) throws IOException {
+        Collection<String> metadataFiles = remoteMetadataDirectory.listFilesByPrefix(
+            MetadataFilenameUtils.METADATA_PREFIX);
+
+        return metadataFiles.stream().filter(
+            file -> (
+                MetadataFilenameUtils.getPrimaryTerm(
+                    file.split(MetadataFilenameUtils.SEPARATOR)) == primaryTerm
+                    && MetadataFilenameUtils.getGeneration(
+                    file.split(MetadataFilenameUtils.SEPARATOR)) == generation)).findAny();
+    }
+
+    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix,
+                         String checksum) throws IOException {
         String remoteFilename;
         if (useCommonSuffix) {
             remoteFilename = dest + SEGMENT_NAME_UUID_SEPARATOR + this.commonFilenameSuffix;
         } else {
             remoteFilename = getNewRemoteSegmentFilename(dest);
         }
         remoteDataDirectory.copyFrom(from, src, remoteFilename, context);
-        String checksum = getChecksumOfLocalFile(from, src);
         UploadedSegmentMetadata segmentMetadata = new UploadedSegmentMetadata(src, remoteFilename, checksum, from.fileLength(src));
         segmentsUploadedToRemoteStore.put(src, segmentMetadata);
     }
+    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+        copyFrom(from, src, dest, context, useCommonSuffix, getChecksumOfLocalFile(from, src));
+    }
+
+    public static long getCommitGenerationFromMdFile(String mdFile) {
```

Comment:
Where is this being used?

Reply:
was using this during snapshot restore, might not be needed now.